### PR TITLE
Return related canonical visualizations on demand

### LINKED ISSUE #12910: Add related canonical visualization to vis endpoint
In order to create the static public_map view, we need to have the related_canonical_visualization information for the "Datasets in use". We have thought that the best option is to add a parameter to the request to add this data (endpoint example: http://username.localhost.lan:3000/api/v1/viz/viz-id)

captura de pantalla 2017-10-05 a las 10 09 17

- Datasets in use (public_map): [https://github.com/CartoDB/cartodb/blob/master/app/views/admin/visualizations/public_map.html.erb#L182](https://github.com/CartoDB/cartodb/blob/master/app/views/admin/visualizations/public_map.html.erb#L182) 
- Visualizations Controller [https://github.com/CartoDB/cartodb/blob/master/app/controllers/admin/visualizations_controller.rb#L274](https://github.com/CartoDB/cartodb/blob/master/app/controllers/admin/visualizations_controller.rb#L274)

### PR DESCRIPTION
**Pull request by juanignaciosl**
This closes #12910.

### CODE REVIEW

File: app/controllers/carto/api/visualization_presenter.rb [hunk 1]
```diff
@@ -70,6 +71,7 @@ def to_poro
         }
 
         poro[:related_tables] = related_tables if related
+        poro[:related_canonical_visualizations] = related_canonical_visualizations if @related_canonical_visualizations
```
Comment (mayoral) [inline app/controllers/carto/api/visualization_presenter.rb]: Can we change the name of the variable to something like `include_related_canonical_viz` so we don't need to refer it with `@` like we do with the other params?

Comment (juanignaciosl) [reply app/controllers/carto/api/visualization_presenter.rb]: Ok, changing the naming so we can use `attr_reader` and avoid the method collision, which is confusing 👍

**Comment by juanignaciosl** [comment]
@elenatorro I've assigned the issue for acceptance to you because I'd like your blessing on this, in order to check if it works for you. Afterwards, @javitonino or @amartinj can do the final acceptance.

**Comment by elenatorro** [comment]
Yes, perfect, I'll tell them once I've tested it. Thank you very much!!! 👍 😍

**Comment by elenatorro** [comment]
Could you tell me the steps to be able to see the related_canonical_visualizations? Because I still can see related_tables in the response, but not the related_canonical_visualizations. I don't know how reproduce this for acceptance :) Thanks in advance!

**Comment by juanignaciosl** [comment]
Do the request, appending fetch_related_canonical_visualizations=true

**Pull request merged**